test/terra/backends/qasm_simulator/qasm_snapshot.py

```diff
@@ -348,154 +350,52 @@ class QasmSnapshotDensityMatrixTests:
-                                                            post_measure=True)
-
-        qobj = assemble(circuits, self.SIMULATOR, memory=True, shots=shots)
-        job = self.SIMULATOR.run(qobj, backend_options=self.BACKEND_OPTS)
-        result = job.result()
-        success = getattr(result, 'success', False)
-        method = self.BACKEND_OPTS.get('method', 'automatic')
-        if method not in QasmSnapshotDensityMatrixTests.SUPPORTED_QASM_METHODS:
-            self.assertFalse(success)
-        else:
-            self.assertTrue(success)
-            self.compare_counts(result,
-                                circuits,
-                                counts_targets,
-                                delta=0.2 * shots)
-            # Check snapshots
-            for i, circuit in enumerate(circuits):
-                data = result.data(circuit)
-                snaps = self.density_snapshots(data, label)
-                for j, mem in enumerate(data['memory']):
-                    target = statevec_targets[i].get(mem)
-                    target = np.outer(target, target.conj())
-                    value = snaps.get(mem)
-                    self.assertTrue(np.allclose(value, target))
+        label = 'density_matrix'
+        snap_qargs = [[0, 1, 2], [0, 2, 1], [1, 0, 2], [1, 2, 0], [2, 0, 1], [2, 1, 0],
+                      [0, 1], [1, 0], [0, 2], [2, 0], [1, 2], [2, 1],
+                      [0], [1], [2]]
+        evolve_qargs = [[0, 1, 2], [0, 2, 1], [1, 0, 2], [2, 0, 1], [1, 2, 0], [2, 1, 0],
+                        [0, 1, 2], [1, 0, 2], [0, 2, 1], [1, 2, 0], [2, 0, 1], [2, 1, 0],
+                        [0, 1, 2], [1, 0, 2], [2, 1, 0]]
+        for squbits, equbits in zip(snap_qargs, evolve_qargs):
+            with self.subTest(msg='qubits={}'.format(squbits)):
+                num_qubits = len(squbits)
+                tmp = circ.copy()
+                tmp.append(Snapshot(label, 'density_matrix', num_qubits), squbits)
+                result = execute(tmp, self.SIMULATOR,
+                                 backend_options=self.BACKEND_OPTS).result()
+                if method not in QasmSnapshotDensityMatrixTests.SUPPORTED_QASM_METHODS:
+                    self.assertFalse(result.success)
+                else:
+                    self.assertSuccess(result)
+                    snapshots = result.data(0)['snapshots']['density_matrix']
+                    value = qi.DensityMatrix(snapshots[label][0]['value'])
+                    target = qi.DensityMatrix.from_label(3 * '0').evolve(circ, equbits)
+                    if num_qubits == 2:
+                        target = qi.partial_trace(target, [2])
+                    elif num_qubits == 1:
+                        target = qi.partial_trace(target, [1, 2])
+                    if value != target:
+                        print('FAILED:', squbits, equbits)
+                        print(value)
+                        print(target)
```

Comment:
Use logging instead of print.

Reply:
These were for debugging, should have been removed